test/DistributionCreator.t.sol

```diff
@@ -38,115 +48,1627 @@ contract DistributionCreatorTest is Fixture {
+//         vm.prank(governor);
+//         creator.setCampaignFees(3, 1e7);
+
+//         IERC20 rewardToken = IERC20(address(angle));
+//         uint256 amount = 100 ether;
+//         uint256 amountAfterFees = 90 ether;
+//         uint32 startTimestamp = uint32(block.timestamp + 600);
+//         uint32 duration = 3600 * 24;
+//         uint32 durationAfterOverride = 3600 * 12;
+
+//         uint256 prevBalance = rewardToken.balanceOf(alice);
+
+//         bytes memory campaignData = abi.encode(
+//             0xbEEfa1aBfEbE621DF50ceaEF9f54FdB73648c92C,
+//             new address[](0),
+//             new address[](0),
+//             "",
+//             new bytes[](0),
+//             new bytes[](0),
+//             hex""
+//         );
+//         vm.prank(alice);
+//         bytes32 campaignId = creator.createCampaign(
+//             CampaignParameters({
+//                 campaignId: bytes32(0),
+//                 creator: alice,
+//                 rewardToken: address(rewardToken),
+//                 amount: amount,
+//                 campaignType: 1,
+//                 startTimestamp: startTimestamp,
+//                 duration: duration,
+//                 campaignData: campaignData
+//             })
+//         );
+
+//         assertEq(rewardToken.balanceOf(alice), prevBalance - amount);
+
+//         vm.warp(block.timestamp + 1000);
+//         vm.roll(4);
+//         // override
+//         vm.prank(alice);
+//         creator.overrideCampaign(
+//             campaignId,
+//             CampaignParameters({
+//                 campaignId: campaignId,
+//                 creator: alice,
+//                 rewardToken: address(rewardToken),
+//                 amount: amountAfterFees,
+//                 campaignType: 3,
+//                 startTimestamp: startTimestamp,
+//                 duration: durationAfterOverride,
+//                 campaignData: campaignData
+//             })
+//         );
+
+//         assertEq(rewardToken.balanceOf(alice), prevBalance - amount - (amountAfterFees * 1e7) / 1e9);
+//     }
+// }
+
+contract UpgradeDistributionCreatorTest is Test, JsonReader {
+    DistributionCreator public distributionCreator;
+    Distributor public distributor;
+    IAccessControlManager public accessControlManager;
+    address public deployer;
+    address public governor;
+    IERC20 public rewardToken;
+    uint256 public chainId;
+    bytes32 public campaignId0;
+    bytes32 public campaignId73;
+    bytes32 public testCampaignId;
+
+    function setUp() public {
+        // Setup environment variables
+        deployer = vm.envAddress("DEPLOYER_ADDRESS");
+        vm.createSelectFork(vm.envString("BASE_NODE_URI"));
+        chainId = block.chainid;
+
+        // Load existing contracts
+        distributor = Distributor(this.readAddress(chainId, "Merkl.Distributor"));
+        distributionCreator = DistributionCreator(this.readAddress(chainId, "Merkl.DistributionCreator"));
```

Comment:
what is this function??

Reply:
utils function to read addresses from sdk registry.